benches/benches/block_target_gas_set/crypto.rs

```diff
@@ -0,0 +1,147 @@
+use crate::{
+    utils::generate_linear_costs,
+    *,
+};
+use rand::{
+    rngs::StdRng,
+    SeedableRng,
+};
+
+// ECK1: Secp251k1 signature recovery
+// ECR1: Secp256r1 signature recovery
+// ED19: edDSA curve25519 verification
+// K256: keccak-256
+// S256: SHA-2-256
+pub fn run_crypto(group: &mut BenchmarkGroup<WallTime>) {
+    let rng = &mut StdRng::seed_from_u64(2322u64);
+
+    let message = Message::new(b"foo");
+
+    let eck1_secret = SecretKey::random(rng);
+    let eck1_signature = Signature::sign(&eck1_secret, &message);
```

Comment:
Hmm, I don't know, but maybe it is also worth testing in case of invalid signatures. Maybe some of these curves work slower in case of invalid signatures.

Or maybe you can find public research regarding the performance of these curves. Based on this research we can select the worst case.

Reply:
I added an invalid test, which seems to be roughly the same efficiency.

I don't know what it would mean to find a worst-case signature. It's possible that some signatures are more expensive to verify than others, but the more you can say about the original message in those cases, the worse the algo is, right? So Idk what we'd be looking for other than finding some specific input that results in a slow test.

Perhaps we could modify the test to just have a bunch of random inputs and see what it averages out to.